build.gradle

```diff
@@ -28,6 +28,54 @@ dependencies {
             "io.dropwizard:dropwizard-testing:$dependencyVersions.dropwizard"
 }
 
+sourceSets {
+    java9 {
+        if (JavaVersion.current() >= JavaVersion.VERSION_1_9) {
+            java {
+                srcDir 'src/main/java9'
+            }
+        }
+    }
+    java10 {
+        if (JavaVersion.current() >= JavaVersion.VERSION_1_10) {
+            java {
+                srcDir 'src/main/java10'
+            }
+        }
+    }
+    java11 {
+        if (JavaVersion.current() >= JavaVersion.VERSION_11) {
+            java {
+                srcDir 'src/main/java11'
+            }
+        }
+    }
+}
+
+compileJava {
+    if(JavaVersion.current() >= JavaVersion.VERSION_1_9) {
```

Comment:
Is there a reason this isn't `VERSION_1_8`?

Reply:
Yes - multi-release jars were introduced in Java 8. If this compiled in Java 8, you only get Java 8 binaries. If you version is at least Java 9, you will get Java 8 binaries as before plus versions-specific binaries for newer Javas. This is the same patters we use in the VSP to ensure RPs can run it on older versions of Java.